File: rust/tcl-registry/src/commands/iapps/tmsh__cd.rs
//! `tmsh::cd` command.
use crate::prelude::*;
const FORMS: &[FormSpec] = &[FormSpec {
    kind: FormKind::Default,
    synopsis: "tmsh::cd <directory>",
}];

pub fn spec() -> CommandSpec {
    CommandSpec {
        name: "tmsh::cd",
        dialects: Some(DialectSet::IAPPS),
        arity: Arity::exact(1),
        hover: Some(HoverSnippet::brief(
            "Changes the current working directory.",
            &["tmsh::cd <directory>"],
            "F5",
        )),
        forms: FORMS,
        ..CommandSpec::DEFAULT
    }
}

File: core/commands/registry/iapps/tmsh__cd.py
"""tmsh::cd -- F5 TMSH directory command."""

from __future__ import annotations

from .._base import CommandDef
from ..models import CommandSpec
from ._base import _tmsh_spec, register


@register
class TmshCd(CommandDef):
    name = "tmsh::cd"

    @classmethod
    def spec(cls) -> CommandSpec:
        return _tmsh_spec(
            "tmsh::cd",
            "Changes the current working directory.",
            "tmsh::cd <directory>",
            min_args=1,
            max_args=1,
        )
